Merge ../custom-templates page into the ../project_templates page

anklesJelled opened the pull request:
TLDR
-----
- Merge the ../advanced/custom-templates page into the ../advanced/project_templates page.
- Expand documentation to cover using a local repo during `init`.
- Re-word documentation where needed.
- Fix typos along the way.

Changes
-----

### ../advanced/project_templates
- Add missing frontmatter
- Reword
- Add advanced/custom-templates content
- Add link to Directory repo

### ../basics/upgrade-v4
- Fixed We're We've typo.



Recommended Future Changes
-----
- advanced/custom-templates (and it's corresponding jp translation) can now be deleted.
- Update basics/upgrade-v4 page to explain the `file://` option.



Changes that should happen even if these changes aren't merged
-----

### ../advanced/project_templates

- Add missing frontmatter.

### ../advanced/custom-templates

- Typo: Thor Templates > First word > Template --> Templates. I think there might be other typos here, I am not familiar enough to know for sure.
- Typo: Template Directory > Last sentence > registed --> registered


### ../basics/upgrade-v4

- Typo: Installing Templates from Git > First Word > We're --> We've
- Installing Templates from Git > I feel that this section is incorrect/misleading because with `file://` you can use a local file template. I would be happy to add/reword this page if you'd like me to.

tdreyno commented:
Looks like this PR has only 1 file. Are those other file changes supposed to be present or will those be additional PRs?

This all makes sense. I'd only add a `redirect` in `config.rb` to handle the removed page and point it to the new one.

anklesJelled commented:
Thanks for the quick response. I just pushed more changes.
Yes, the first commit only changed one file. I wanted to see what your thoughts were before changing more.
I did NOT redirect the Japanese version of the deleted file in config.rb. I wasn't sure if that was necessary. Any more suggestions?

tdreyno commented on config.rb:
```diff
@@ -17,6 +17,7 @@
   "basics/helpers.html"                  => "basics/helper_methods.html",
   "basics/getting-started.html"          => "basics/install.html",
   "basics/asset-pipeline.html"           => "advanced/asset_pipeline.html",
+  "advanced/custom-templates/index.html" => "advanced/project_templates/"
```
I think these need to be the non-directory versions. `advanced/custom-templates.html -> advanced/project_templates.html`

tdreyno commented:
These are great. I'd suggest not removing the japanese version, our translator can handle that change when he gets a chance.

1 comment in the redirect code.

Great changes, keep them coming :)

anklesJelled commented:
Thank you for the compliment. It is about the least I could do... :) Anyway, I

- restored the jp translation of the now-deleted page
- changed the redirect in config.rb

I think this is good to go...?

tdreyno commented:
Looks good. I'll merge tomorrow at work :)

anklesJelled commented:
I just want to make sure that you aren't waiting on me for something...?

tdreyno commented:
Thanks for bumping this. I forgot, I was out at a conference :)
